src/lib/fetch.ts

```diff
@@ -52,7 +39,12 @@ async function _handleRequest(
         return result.json()
       })
       .then((data) => resolve(data))
-      .catch((error) => handleError(error, reject))
+      .catch(() =>
+        reject({
+          message: NETWORK_FAILURE,
```

Comment:
Since we are throwing even if `result.ok` is false, lets have a generic message like "Request Failed"? Cos this will get triggered both for network errors and server errors like timeouts etc